binder/views.py

```diff
@@ -2870,15 +2870,21 @@ def dispatch_file_field(self, request, pk=None, file_field=None):
 
 		file_field_name = file_field
 		file_field = getattr(obj, file_field_name)
+		field = self.model._meta.get_field(file_field_name)
 
 		if request.method == 'GET':
 			if not file_field:
 				raise BinderNotFound(file_field_name)
 
-			guess = mimetypes.guess_type(file_field.path)
-			guess = guess[0] if guess and guess[0] else 'application/octet-stream'
+			guess = mimetypes.guess_type(file_field.name)
+			content_type = (guess and guess[0]) or 'application/octet-stream'
+			serve_directly = isinstance(field, BinderFileField) and field.serve_directly
 			try:
-				resp = StreamingHttpResponse(open(file_field.path, 'rb'), content_type=guess)
+				if serve_directly:
+					resp = HttpResponse(content_type=content_type)
+					resp['X-Accel-Redirect'] = '/internal/media/' + file_field.name
```

Comment:
Do we want to replace `/internal/media/` with a ENV setting which has this as default?
Same with `X-Accel-Redirect`, that could be a image redirect header setting with a default?

Reply:
The header is very specific to this functionality so I don't think it makes sense to parameterise it. Put differently, could that variable ever be set to anything other than `X-Accel-Redirect` such that it does the same thing?

Reply:
I suspect `X-Sendfile` for [apache2](https://www.yiiframework.com/wiki/129/x-sendfile-serve-large-static-files-efficiently-from-web-applications) and [User defined](https://caddyserver.com/docs/json/apps/http/servers/routes/handle/reverse_proxy/handle_response/) headers and logic in Caddy

Reply:
Very well.